genesis-programs/src/lib.rs

```diff
@@ -95,6 +95,14 @@ pub fn get_entered_epoch_callback(operating_mode: OperatingMode) -> EnteredEpoch
             bank.epoch(),
             operating_mode
         );
+        if operating_mode == OperatingMode::Preview && bank.epoch() == 50 {
```

Comment:
Placing this at `get_entered_epoch_callback` is nice idea. Well, I didn't notice it when working #9527 to be honest while I remotely remembered this mechanism.  ;)

However, because `incinerator_enabled` is [`#[serde(skip)]`](https://github.com/solana-labs/solana/pull/9836/files#diff-a7549f152920d85fb44e6a784b8e5e1fR356), I don't think we can use it for this. that callback can be used for changing state of non-`#[serde(skip)]` `struct` member fields including `incinerator_enabled` (which would break abi).

Specifically, we need to ensure to initialize correctly `bank.incinerator_enabled` immediately after snapshot loading.

Also, the condition must be `>=`.

Say,  we're at the epoch 53 and a validator loaded a snapshot of epoch 53. Then according to `Bank::default()`, `bank.incinerator_enabled` is `false` immediately after auto-derived `Bank::deserialize()`. And, none of this callback and this condition won't trigger. So, indeterminism is introduced to the cluster.

You also can refer to my way of gating. :) https://github.com/solana-labs/solana/pull/9527/files#diff-a7549f152920d85fb44e6a784b8e5e1fR506 Note: I haven't conducted any rigorous testing yet.. Maybe time to do it for me. :)